lib/Lift.cpp

```diff
@@ -546,57 +546,55 @@ CreateConstFromMemory(const uint64_t addr, llvm::Type *type,
     } break;
 
     case llvm::Type::PointerTyID: {
+      const auto pointer_type = llvm::dyn_cast<llvm::PointerType>(type);
+      const auto size = dl.getTypeSizeInBits(type);
+      auto val = ReadValueFromMemory(addr, size, arch, program);
+      result = llvm::Constant::getIntegerValue(pointer_type, val);
     } break;
 
     case llvm::Type::StructTyID: {
+
       // Take apart the structure type, recursing into each element
       // so that we can create a constant structure
-      auto struct_type = llvm::dyn_cast<llvm::StructType>(type);
-
-      auto num_elms = struct_type->getNumElements();
+      const auto struct_type = llvm::dyn_cast<llvm::StructType>(type);
+      const auto num_elms = struct_type->getNumElements();
+      std::vector<llvm::Constant *> initializer_list;
       auto elm_offset = 0;
 
-      std::vector<llvm::Constant *> const_list;
-
       for (std::uint64_t i = 0U; i < num_elms; ++i) {
-        auto elm_type = struct_type->getElementType(i);
-        auto elm_size = dl.getTypeSizeInBits(elm_type);
+        const auto elm_type = struct_type->getElementType(i);
+        auto elm_size = dl.getTypeAllocSize(elm_type);
```

Comment:
Pulling structure elements out of memory is a bit more subtle than iterating over the structure types themselves. Can you change this code to use `dl.getStructLayout`, and then follow a similar API usage/structure as you see here: https://github.com/lifting-bits/remill/blob/master/lib/BC/Util.cpp#L1578-L1597

Reply:
Updated it to use the `StructLayout` and use the struct type API's to get elements.